### PR TITLE
[RESTEASY-2179] updated arquillian versions;  Added security manage c…

### PR DESCRIPTION
**Pull request by rsearls**
…heck code blocks where possible.  Add permission classes to test archives as needed

### CODE REVIEW

File: testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/microprofile/restclient/RestClientProxyTest.java [hunk 1]
```diff
@@ -28,6 +29,7 @@
 import io.reactivex.Single;
 
 @RunWith(Arquillian.class)
+@RunAsClient
```
Comment (asoldano) [inline testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/microprofile/restclient/RestClientProxyTest.java]: Why is this needed? I mean, it's not a problem, just wondering why.

Comment (rsearls) [reply testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/microprofile/restclient/RestClientProxyTest.java]: It is needed when security manager is being used.  Without it arquillian throws exception,
java.util.NoSuchElementException

**Comment by rsearls** [comment]
Rebase complete

File: server-adapters/resteasy-netty4-cdi/pom.xml [hunk 2]
```diff
@@ -14,7 +14,7 @@
     <description/>
     <properties>
         <dep.arq.weld.se>2.0.0.Final</dep.arq.weld.se>
-        <dep.arq.version>1.1.15.Final</dep.arq.version>
+       <dep.arq.version>1.4.1.Final</dep.arq.version>
```
Comment (asoldano) [inline server-adapters/resteasy-netty4-cdi/pom.xml]: uh, just spotted this. Is there a reason why we can't use the 'dep.arquillian-bom.version' from the resteasy-dependencies-bom/pom.xml? (to avoid duplicating the property)

Comment (rsearls) [reply server-adapters/resteasy-netty4-cdi/pom.xml]: rebase .. complete
version ref change .. complete
